src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -427,6 +427,10 @@ public override async ValueTask<IReadOnlyList<BulkStateItem>> GetBulkStateAsync(
                 return invokeResponse;
 
             }
+            catch (RpcException rpc) when (rpc.StatusCode == StatusCode.Cancelled && cancellationToken.IsCancellationRequested)
```

Comment:
Why are you doing this instead of setting `GrpcChannelOptions.ThrowOperationCanceledOnCancellation` to true?

Reply:
discussed with @rynowak again and used the solution that you suggested.